internal/manifest/skrresources/manifestcollector.go

```diff
@@ -0,0 +1,60 @@
+package skrresources
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/manifest/manifestclient"
+)
+
+// ManifestLogCollector is a collector for remote Manifest objects. It delegates the calls to the embedded generic collector if collection is enabled for the given Manifest.
+type ManifestLogCollector struct {
+	collector *LogCollector
+	enabled   bool
+}
+
+func NewManifestLogCollector(manifest *v1beta2.Manifest, owner client.FieldOwner) *ManifestLogCollector {
+	key := ""
+	enabled := false
+	if manifest != nil {
+		key = string(manifest.GetUID())
+		enabled = isManifestCollectionEnabled(manifest)
+	}
+	return &ManifestLogCollector{
+		collector: NewLogCollector(key, manifestclient.DefaultFieldOwner),
+		enabled:   enabled,
+	}
+}
+
+// Implements the skrresources.ManagedFieldsCollector interface.
+func (c *ManifestLogCollector) Collect(ctx context.Context, obj client.Object) {
+	if c.enabled {
+		c.collector.Collect(ctx, obj)
+	}
+}
+
+// Implements the skrresources.ManagedFieldsCollector interface.
+func (c *ManifestLogCollector) Emit(ctx context.Context) error {
+	if c.enabled {
+		return c.collector.Emit(ctx)
+	}
+	return nil
+}
+
+// isCollectionEnabled checks if managed fields detection is enabled for the given manifest.
+// The detection is enabled by default, but can be controlled by setting a specific label on the manifest CR.
+func isManifestCollectionEnabled(obj *v1beta2.Manifest) bool {
+	if obj == nil {
+		return false
+	}
+
+	detectionLabelName := getManagedFieldsAnalysisLabel()
+	if detectionLabelName == "" {
+		return true
```

Comment:
Just wondering, should we rather turn it off by default? Otherwise we will directly start to collect the info for all manifests on the given landscape. Its probably better if we just start with collecting it selectively?

Reply:
You got a point. What I wanted was some way to easily collect the data from all manifests. But I agree, that making it default is probably too much.
How about this:
- disabled when empty string
- enabled selectively when the configured label exists
- enabled for all when the configured label exists with a specific value (e.g: configuredLabelName=ALL)

What do you think?

Reply:
This would work if it was an ENV var, but since this is a label on a single manifest it doesn't make sense to add an `ALL` entry?

I actually don't think we will want to collect it for all anyway. We want to collect it for a relatively small number (e.g., 10 clusters) so we can detect patterns (e.g., KEB always putting a label on kyma-system namespace). So it should be fine by selective only I think.

Reply:
👍